### PR TITLE
[influxdb] Add compatibility with InfluxDB Cloud Serverless 

### PR DESCRIPTION
**Pull request by PLeusmann**
Update influx client to 4.3.0 and use ping endpoint to check for database availability. Fixes https://github.com/openhab/openhab-addons/issues/16147

Chose not to go to newest client version since this will resolve to some vulnerabilities in the dependency tree according to maven central (https://mvnrepository.com/artifact/com.squareup.retrofit2/adapter-rxjava3/2.9.0)

Did not run all tests since they seem to require some running influx instances and i could not find information on how to setup those. 

Resulting jar: [org.openhab.persistence.influxdb-4.2.0-SNAPSHOT.jar.zip](https://github.com/openhab/openhab-addons/files/13798883/org.openhab.persistence.influxdb-4.2.0-SNAPSHOT.jar.zip)

### CODE REVIEW

File: bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/influx2/InfluxDB2RepositoryImpl.java [hunk 1]
```diff
@@ -117,8 +117,16 @@ public void disconnect() {
     public boolean checkConnectionStatus() {
         final InfluxDBClient currentClient = client;
         if (currentClient != null) {
+            boolean isUp = false;
             Ready ready = currentClient.ready();
-            boolean isUp = ready != null && ready.getStatus() == Ready.StatusEnum.READY;
+            if (ready != null) {
+                isUp = ready.getStatus() == Ready.StatusEnum.READY;
+            } else {
+                logger.warn(
```
Comment (J-N-K) [inline bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/influx2/InfluxDB2RepositoryImpl.java:125]: If it's "normal", it should only be logged at DEBUG level.

**Review by J-N-K — CHANGES_REQUESTED** [review]
Otherwise LGTM, what do you mean by "did not run all tests"?

**Comment by PLeusmann** [comment]
> Otherwise LGTM, what do you mean by "did not run all tests"?

Probably I misinterpreted. Test runs raise a lot of stack traces such as `[main] WARN  o.o.p.i.i.i.InfluxDB1RepositoryImpl - database error: java.net.ConnectException: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:8086`. I guessed there are some integration tests requiring influxdb-instances. This seems not to be the case. 

Tests still seem to pass, though.

**Comment by PLeusmann** [comment]
@J-N-K I made the changes you requested.

**Review by J-N-K — APPROVED** [review]
LGTM

**Pull request merged**